hack/verify/mdlint.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# script to run markdown linters
+set -o errexit -o nounset -o pipefail
+
+# cd to the repo root
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
+cd "${REPO_ROOT}"
+
+docker run --rm -v "$(pwd)":/build \
+  gcr.io/cluster-api-provider-vsphere/extra/mdlint:0.17.0
```

Comment:
is this URL going to be supported or should we try to find a more common docker image?

Reply:
The docker image maintained by @akutz currently.

Use this image just because it is currently included in the [kubernetes-sigs repository](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/hack/tools/mdlint/Dockerfile) (also a member of the official community)

@akutz PTAL? Thanks.